## Задача 1

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания mu с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

## Решение
Воспользуемся формулой
$T = \dfrac{\overline{X} - \mu}{\sigma / \sqrt{n}}$, cтатистика $T$ имеет стандартное распределение

In [10]:
import numpy as np
from scipy import stats

sig=16
n=256
x=80
p=0.95
alpha=1-p

t = stats.norm.ppf(1 - alpha/2)
print(t)

1.959963984540054


In [11]:
print("Доверительный интервал: ", x - t * sig / np.sqrt(n), x + t * sig / np.sqrt(n))

Доверительный интервал:  78.04003601545995 81.95996398454005


## Задача 2

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. 
1. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? 
2. Найдите P-значение для данного теста.

## Решение
Гипотеза Н0="средний вес пачки печенья составляет 200". Альтернативная гипотеза "средний вес пачки печенья не 200".
Т.к. мы  не знаем дисперсию случайной величины, воспользуемся формулой:
$$t = \dfrac{\overline{X} - \mu}{\sigma_X / \sqrt{n}},$$

In [3]:
cookies = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

mean = cookies.mean()
std = cookies.std(ddof=1)

mean, std

(198.5, 4.453463071962462)

In [4]:
t = (mean-200)/(std/np.sqrt(cookies.shape[0]))
t

-1.0651074037450896

In [5]:
alpha = 0.01

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-2.595246484783763, 2.5952464847837624)

In [6]:
# критическая область
resualt = "Гипотеза верна"
for i in (0.01, 0.05, 0.1):
    t1 = stats.t.ppf(i / 2, df=n - 1)
    t2 = stats.t.ppf(1 - i / 2, df=n - 1)
    if t<t1 or t>t2:
        resualt="Гипотеза не верна"
    print(f"a={i}, t1= {t1}, t2= {t2}, попадает в критичесвкую область: {resualt}")

a=0.01, t1= -2.595246484783763, t2= 2.5952464847837624, попадает в критичесвкую область: Гипотеза верна
a=0.05, t1= -1.9693105698491933, t2= 1.9693105698491928, попадает в критичесвкую область: Гипотеза верна
a=0.1, t1= -1.6508510924984823, t2= 1.6508510924984816, попадает в критичесвкую область: Гипотеза верна


Утверждение продавца не противоречиво.

## Задача 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

# Решение
Гипотеза Н0 = "шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм"

Используем формулу 
$Z = \dfrac{\overline{X} - \mu}{\sigma / \sqrt{n}},$

In [7]:
n = 100
mean=17.5
alpha = 0.025
disp = 4
mu=17

In [8]:
# НАйдем критическую область. Т.к. правосторонняя, то будет от t до infinity

t = stats.norm.ppf(1 - alpha)
t

1.959963984540054

In [9]:
Z = (mean-mu)/(np.sqrt(disp)/np.sqrt(n))
Z

2.5

Так как критическая область правосторонняя и 2.5 > 1.95 (т.е. лежит правее 1.95), то гипотезу Н0 отвергаем.